### PR TITLE
Remove usless test and resolved comments

### LINKED ISSUE #15815: RBAC Migration Followup #3: Test Fixes
- [x] Confirm changes to `TestAuthorizationResolution` and `TestAuthorizationResourceAccessReview` in `test/integration/authorization_test.go`
- [x] Remove `TestPolicyCache`

### PR DESCRIPTION
**Pull request by simo5**
Fixes  #15815

### CODE REVIEW

File: test/integration/authorization_test.go [hunk 1]
```diff
@@ -302,14 +302,15 @@ func TestAuthorizationResolution(t *testing.T) {
 		t.Fatalf("unexpected error: %v", err)
 	}
 
-	// try to add Valerie to a non-existent role, looping until it is true due to
-	// the policy cache taking time to react
+	// try to add Valerie to a non-existent role, looping until it is done.
+	// If a Not Found error is raised, simply retry as this is likely due to
+	// the role cache taking time to react
```
Comment (enj) [inline test/integration/authorization_test.go]: I think in general this wait no longer makes any sense (there is no role cache).

```go
if err := addValerie.AddRole(); err != nil {
    t.Fatalf("unexpected error: %v", err)
}
```

@deads2k any thoughts?

File: test/integration/authorization_test.go [hunk 2]
```diff
@@ -318,7 +319,7 @@ func TestAuthorizationResolution(t *testing.T) {
 	roleWithGroup.Name = "with-group"
 	roleWithGroup.Rules = append(roleWithGroup.Rules, authorizationapi.PolicyRule{
 		Verbs:     sets.NewString("list"),
-		Resources: sets.NewString("builds"), // TODO we may need to track these down
+		Resources: sets.NewString("builds"),
```
Comment (enj) [inline test/integration/authorization_test.go]: @deads2k do we care that this used to be `Resources: sets.NewString("resourcegroup:builds")`?

**Comment by simo5** [comment]
Still seeing this flake even if it has been closed #14575 
/retest

**Comment by simo5** [comment]
@deads2k can you comment on @enj's question ?

File: test/integration/authorization_test.go (hunk 2 above)
Comment (deads2k) [reply test/integration/authorization_test.go]: > @deads2k do we care that this used to be Resources: sets.NewString("resourcegroup:builds")?

For the test?  No.

**Pull request merged**